### PR TITLE
Deal with patients and specialist records separately in the address book

### LINKED ISSUE #5: As a GP Clinic Management Staff I can add a new patient to the records
... so that I can keep track of the details of new patients

### LINKED ISSUE #6: As a GP Clinic Management Staff I can add a new specialist to the records
... so that I can keep track of details of new specialists patients can be referred to.

### LINKED ISSUE #38: As a GP clinic management staff, I can deal with patient and specialist records separately
...so that I can easily do patient-specific or specialist-specific operations

### PR DESCRIPTION
**Pull request by vansh284**
This PR introduces changes that makes the parser enforce the use of `-pa` or `-sp` tags in order to add patients or specialists into the address book for the relevant commands.
In most cases in the address book, wherever any class related to a `Person` existed, this class has been made abstract and two concrete subclasses `Patient` and `Specialist`. 
Subsequent patient-specific and specialist-specific behavior should be added to these sub-classes for intended behavior.
A UI feature to help distinguish between patients and specialists in the displayed list has also been added (this should be temporary, it isn't the prettiest).

Commands known to still not work fully as intended: `list`, `edit`, `find`, `delete`. In particular, these commands ignore the 
`-pa`/`-sp` tags. Changes to the `ModelManager` class will likely be needed to make them work as intended (e.g. `list -pa` should only list the patients, not everyone).

Closes #38, closes #5 and closes #6.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java
Comment (Kokseng1) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:1]: LGTM

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java
Comment (Kokseng1) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:1]: LGTM

**Review by Kokseng1 — COMMENTED** [review]
Reviewed some files, all LGTM. Lets all continue reviewing the rest of the PR

**Review by alvinlim277 — APPROVED** [review]
LGTM! Really thorough work in combing through and editing the existing codebase to fit the new Patient and Specialist classes.

**Pull request merged**